Doc/c-api/typeobj.rst

```diff
@@ -1034,7 +1034,8 @@ and :c:data:`PyType_Type` effectively act as defaults.)
       the type, and the type object is INCREF'ed when a new instance is created, and
       DECREF'ed when an instance is destroyed (this does not apply to instances of
       subtypes; only the type referenced by the instance's ob_type gets INCREF'ed or
-      DECREF'ed).
+      DECREF'ed). Heap types must also :ref:`support garbage collection <supporting-cycle-detection>`
```

Comment:
Let's soften the language. It's technically possible to create a type that doesn't form reference cycles, if you're really determined. (AFAIK: it'd have to be an immutable non-subclassable type with no associated module and with instances whose attributes can't form cycles.)
```suggestion
      DECREF'ed). Heap types should also :ref:`support garbage collection <supporting-cycle-detection>`
```